gpkit/nomials.py

```diff
@@ -590,11 +580,18 @@ class MonoEQConstraint(Constraint):
     A Constraint of the form Monomial == Monomial.
     Stored internally as a Monomial Constraint, (1 == self).
     """
-    def _set_operator(self, p1, p2):
+    def __init__(self, m1, m2):
+        super(MonoEQConstraint, self).__init__(m1, m2)
+        self.__class__ = MonoEQConstraint  # todo should not have to do this
         self.oper_l = " = "
         self.oper_s = " == "
-        self.leq = Constraint(p2, p1)
-        self.geq = Constraint(p1, p2)
+        # next two lines would be clean implementation, but Constraint
+        # inheritance from Posy won't allow division
```

Comment:
I was just trying to wrap my head around the object model. It's a little confusing that MonoEQConstraint, having inherited from Constraint, IS a Constraint, but also stores Constraint objects in the attributes `self.leq` and `self.geq`.

We could certainly switch this back to initializing `Constraint` objects, but that means `Constraint.__init__` gets called 3 times for every `MonoEQConstraint` we create.

We could also consider removing `self.leq` and `self.geq`, and handling that only when we build a GP (i.e. add the constraint and its inverse).

Ideas welcome - none of this is a big deal, just seemed a little awkward.